python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -309,14 +287,22 @@ def __hash__(self):
         return hash(tuple(self.__repr__()))
 
     @functools.lru_cache(maxsize=100)
-    def _time_window_for_partition_key(self, *, partition_key: str) -> TimeWindow:
+    def time_window_for_partition_key(self, partition_key: str) -> TimeWindow:
         partition_key_dt = pendulum.instance(
             datetime.strptime(partition_key, self.fmt), tz=self.timezone
         )
+        # for simple cases, no need to invoke iterator
+        if self.is_basic_hourly:
+            return TimeWindow(partition_key_dt, partition_key_dt.add(hours=1))
+        if self.is_basic_daily:
+            return TimeWindow(partition_key_dt, partition_key_dt.add(days=1))
```

Comment:
do we need to worry about timezone stuff in this case? cc @gibsondan

Reply:
oof. This is the relevant bit that could come into play when adding a day https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_utils/schedules.py#L157-L168

Wouldn't be an issue in UTC or if the hour is 0